### ISSUE TITLE
Possible feature: closure for results of insert statement

### ISSUE THREAD

**Issue by zathras** [JetBrains/Exposed#808]
What if you allowed a closure to be passed to insert to get the value(s) of the column(s) that make up the primary key?  That is, instead of:
```
val pragueId = Cities.insert {
            it.update(name, stringLiteral("   Prague   ").trim().substring(1, 2))
        }[Cities.id]
```
what if there were the option of, say:
```
val pragueId = Cities.insert {
            it.update(name, stringLiteral("   Prague   ").trim().substring(1, 2))
        } getKey { it[id] }
```
`getKey` would be an inline function taking a code block with a receiver type, so that fields could be referenced.  Two nice things about this syntax:

*  It's more like the select statement, where the getters are in a lambda-with-receiver
*  If you have more than one column in the primary key, you can naturally use a destructuring assignment, without any intermediate variables to hold the thing that holds the `ResultSet`

Just thought I'd share this idea.  It occurred to me when making a much less ambitious Kotlin/SQL library, https://db9010.jovial.com/, where the equivalent would be:
```
val pragueId = Cities.run {
    myDatabase.insertInto(this) {
        it[name] = "Prague"
    } run {
        it[id]
    }
}
```
It kind of bothered me that the way you address column values after an insert wasn't harmonious with the rest of Exposed.

**Comment by Tapac** [comment]
Hi, you could use `let` to achieve a similar functionality:
```kotlin
Cities.insert {
    it[name] = stringLiteral("   Prague   ").trim().substring(1, 2)
}.let {
   it[City.id]
}
```
Or you want to get `City` as `this` in the closure scope?

**Comment by zathras** [comment]
Yes, `City` as `this` in the closure scope.  The point is that it's a little jarring how `City` is in the closure scope in the "body" of the insert, but it's not when you get the results.  That means that you can say `id` in one place, but you're forced to say `City.id` in the other; this interferes with the goal of making the Kotlin code "feel" like a complete, consistent SQL-friendly language.

It's a minor point, but one that can be easily resolved with an inline function.
